src/bgw/job.c

```diff
@@ -451,6 +453,24 @@ ts_bgw_job_permission_check(BgwJob *job)
 				(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
 				 errmsg("insufficient permssions to alter job %d", job->fd.id)));
 }
+
+void
+ts_bgw_job_validate_job_owner(Oid owner, JobType type)
+{
+	HeapTuple role_tup = SearchSysCache1(AUTHOID, ObjectIdGetDatum(owner));
+	Form_pg_authid rform = (Form_pg_authid) GETSTRUCT(role_tup);
+
+	if (!rform->rolcanlogin)
+	{
+		ReleaseSysCache(role_tup);
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION),
+				 errmsg("permission denied to start %s background process", job_type_names[type]),
+				 errhint("Hypertable owner must have LOGIN permission to run background tasks.")));
```

Comment:
Minor: Supporting this is easier if the role name is shown in the error message.